datasets/trecqa.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+import torch
+from torchtext.data.dataset import Dataset
+from torchtext.data.example import Example
+from torchtext.data.field import Field
+from torchtext.data.iterator import BucketIterator
+from torchtext.vocab import Vectors
+
+from datasets.idf_utils import get_pairwise_word_to_doc_freq, get_pairwise_overlap_features
+
+
+class TRECQA(Dataset):
+    NAME = 'trecqa'
+    NUM_CLASSES = 2
+    ID_FIELD = Field(sequential=False, tensor_type=torch.FloatTensor, use_vocab=False, batch_first=True)
+    TEXT_FIELD = Field(batch_first=True, tokenize=lambda x: x)  # tokenizer is identity since we already tokenized it to compute external features
```

Comment:
The better way of doing this is:
```python
TEXT_FIELD = Field(batch_first=True, sequential=False)
```
Refer torchtext [doc](https://github.com/pytorch/text/blob/master/torchtext/data/field.py#L75)

Reply:
You are right in that this is already tokenized, so no tokenization is required.

> sequential: Whether the datatype represents sequential data. If False,
            no tokenization is applied. Default: True.

When I tried to change this to `sequential=False`, I get a `TypeError: unhashable type: 'list` on the last line in this block from `torchtext` [source](https://github.com/pytorch/text/blob/master/torchtext/data/field.py#L237-#L241):

```python
        for data in sources:
            for x in data:
                if not self.sequential:
                    x = [x]
                counter.update(x)
```
As you can see, if data is not declared sequential, then torchtext wraps data in another list. I already have a list so this will create a list of list, which is incorrect.This is sequential data (a list of tokens), so I think what I did originally makes sense. It's just that the sequence data is already tokenized.

Reply:
I'm sorry, you're right. This was not the case with the previous version of torchtext.